### ISSUE TITLE
Send an email notification on lambda execution error

### ISSUE THREAD

**Issue by GabLeRoux** [GabLeRoux/ssl-expiry-check-terraform#1]
So this is working, deployment works fine 👍 but I still need to figure out how to setup the API Gateway to catch 400 errors and trigger an SNS aws notification so I can actually know if something went wrong.

Related documentation for API Gateway:  
https://www.terraform.io/docs/providers/aws/r/api_gateway_rest_api.html

Related documentation for cert error catching:  
https://github.com/LucasRoesler/ssl-expiry-check#aws-api-gateway

**Comment by GabLeRoux** [comment]
I thought a bit more about it and I don't think API Gateway is what I'm looking for.

Here's what I think I need:
1. A Cloudwatch alert configured to trigger an SNS notification on lambda error
2. Another lambda function that will send a preformated email with the desired data
3. A way to trigger the new lambda function with the error data

Some related articles and QAs:

* [Filter and Pattern Syntax](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html)
  unless I simply use the exit code and output. If everything is fine, current lambda exits with exit code 0. In case of any error, it will exit with -1 and prints a json (see json example).
* [How do I Create a CloudWatch Email Alert With Custom Content?](https://stackoverflow.com/questions/44936618/how-do-i-create-a-cloudwatch-email-alert-with-custom-content)
  tldr; Create a Lambda on cloudwatch alarm that sends an email using _SES SMTP credentials_ to send a formatted email with the _alarm trigger event data_.
* [How to send custom alerts or notifications](https://forums.aws.amazon.com/thread.jspa?threadID=148832)
  tldr;
  1. Have the Cloudwatch alert publish the alarm to a SNS topic, [here's an example for an alert on CPU util](http://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/US_AlarmAtThresholdEC2.html)
  2. Have an [HTTP endpoint subscribed to the SNS topic to receive the alarm](http://docs.aws.amazon.com/sns/latest/dg/SendMessageToHttp.html)
  3. Have the HTTP endpoint create the mail message in the format you require, based on the information passed from SNS, and email it to your support / alert email. Unfortunately we have no examples of this, but some custom code on the HTTP endpoint should be able to do that for you.

## cloudwatch json message example (found in the logs)

Found in cloudwatch:

```json
{
  "message": "Cert Errors",
  "results": [
    "gableroux.com cert is fine",
    "example.com ssl error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:749)",
    "wronghostname.example.com cert error hostname 'wronghostname.example.com' doesn't match either of 'a.example.com', 'b.example.com', 'c.example.com'",
    "domain_not_found.example.com socket error: [Errno -2] Name or service not known"
  ],
  "additional_hosts": [
    ""
  ]
}
```

## Example email

I'd like to have this sent to my inbox when something goes wrong:

```html
Hi,  
SSLExpiry found the following problems:
<ul>
  <li><a href="example.com">example.com</a> ssl error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:749)</li>
  <li><a href="wronghostname.example.com">wronghostname.example.com</a> cert error hostname 'wronghostname.example.com' doesn't match either of 'a.example.com', 'b.example.com', 'c.example.com'</li>
  <li><a href="domain_not_found.example.com">domain_not_found.example.com</a> socket error: [Errno -2] Name or service not known</li>
</ul>

<ul>
  <li><a href="link-to-cloudwatch-event">Cloudwatch event</a></li>
  <li><a href="{{app.run_url}}">Check again</a></li>
</ul>

With love, <a href="link-to-SSLExpiry-service">your SSLExpiry service</a>
```

Actua jinja2 template (maybe):

```jinja2
Hi,  
{{app.name}} found the following problems:
<ul>
  {% for result in results %}
  <li><a href="https://{{result.domain}}/">{{result.domain}}</a> {{error.message}}</li>
</ul>

<ul>
  <li><a href="{{cloudwatch.event.url}}">Cloudwatch event</a></li>
  <li><a href="{{app.run_url}}">Check again</a></li>
</ul>

With love, <a href="{{app.service_url}}">your {{app.name}} service</a>
```
